common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp

```diff
@@ -409,6 +466,131 @@ bool mon_wlan_hal_dwpal::sta_link_measurements_11k_request(const std::string &st
     return true;
 }
 
+bool mon_wlan_hal_dwpal::dcs_scan_trigger(int dwell_time_msec,
+                                          const std::vector<unsigned int> &channel_pool)
+{
+    LOG(DEBUG) << __func__ << " received on interface=" << m_radio_info.iface_name;
+
+    //build scan parameters
+    ScanParams dcs_scan_params;
+    memset((void *)&dcs_scan_params, 0, sizeof(dcs_scan_params)); //must to ensure freq are zero
+    size_t out_len                                = 0;
+    unsigned char out_data[NL_MAX_REPLY_BUFFSIZE] = {'\0'};
```

Comment:
Why is this needed? you provide the buffer length so why should you set null termination explicitly?

Reply:
length is not provided, its an output parameter in order to know how much bytes reply_buffer contains.